tests/test_zookeeper.py

```diff
@@ -1,137 +1,90 @@
-            ('zookeeper.nodes',                    487L),
-        ]
-
-
-        with patch.object(zk, '_send_command', send_command_gte_v344):
-            zk.check(instances[0])
-        service_checks = zk.get_service_checks()
-        self.assertEquals(len(service_checks), 2)
-        self.assertEquals(service_checks[0]['check'], 'zookeeper.ruok')
-        self.assertEquals(service_checks[0]['status'], AgentCheck.WARNING)
-        self.assertEquals(service_checks[1]['check'], 'zookeeper.mode')
-        self.assertEquals(service_checks[1]['status'], AgentCheck.OK)
-
-        metrics = zk.get_metrics()
-        self.assertEquals(metrics[0][3]['tags'], ['mode:leader'])
-        self.assertEquals(sorted([(name, val) for name, _, val, _ in metrics]), sorted(expected))
+from tests.common import AgentCheckTest
+
+
+@attr(requires='zookeeper')
+class ZooKeeperTestCase(AgentCheckTest):
+    CHECK_NAME = 'zk'
+
+    CONFIG = {
+        'host': "127.0.0.1",
+        'port': 2181,
+        'expected_mode': "standalone",
+        'tags': []
+    }
+
+    WRONG_EXPECTED_MODE = {
+        'host': "127.0.0.1",
+        'port': 2181,
+        'expected_mode': "follower",
+        'tags': []
+    }
+
+    CONNEXION_FAILURE_CONFIG = {
+        'host': "127.0.0.1",
+        'port': 2182,
+        'expected_mode': "follower",
+        'tags': []
+    }
+
+    METRICS = [
+        'zookeeper.latency.min',
+        'zookeeper.latency.avg',
+        'zookeeper.latency.max',
+        'zookeeper.bytes_received',
+        'zookeeper.bytes_sent',
+        'zookeeper.connections',
+        'zookeeper.connections',
+        'zookeeper.bytes_outstanding',
+        'zookeeper.outstanding_requests',
+        'zookeeper.zxid.epoch',
+        'zookeeper.zxid.count',
+        'zookeeper.nodes',
+    ]
+
+    def test_check(self):
+        """
+        Collect ZooKeeper metrics.
+        """
+        config = {
+            'instances': [self.CONFIG]
+        }
+        self.run_check(config)
+
+        # Test metrics
+        for mname in self.METRICS:
+            self.assertMetric(mname, tags=['mode:standalone'], count=1)
+
+        # Test service checks
+        self.assertServiceCheck("zookeeper.ruok", status=AgentCheck.OK)
+        self.assertServiceCheck("zookeeper.mode", status=AgentCheck.OK)
+
+        self.coverage_report()
+
+    def test_wrong_expected_mode(self):
+        """
+        Raise a 'critical' service check when ZooKeeper is not in the expected mode
+        """
+        config = {
+            'instances': [self.WRONG_EXPECTED_MODE]
+        }
+        self.run_check(config)
+
+        # Test service checks
+        self.assertServiceCheck("zookeeper.mode", status=AgentCheck.CRITICAL)
+
+    def test_error_state(self):
+        """
+        Raise a 'critical' service check when ZooKeeper is in an error state
+        """
+        config = {
+            'instances': [self.CONNEXION_FAILURE_CONFIG]
+        }
+        self.run_check(config)
```

Comment:
I would expect an exception here. The check is not raising an exception when it can't connect to zookeeper ? 
It should, can you fix that if it's not the case ?